### ISSUE TITLE
DCA Cooldown

### ISSUE THREAD

**Issue by jessmil** [taniman/profit-trailer#331]
Hey, 
we spoke about this some days ago, but it was not listed here.

I would like to have a DCA Cooldown = x ( in mins)

We often have the problem that dca buys several times within 1 or 2 Minutes and the dca level is already @ 5 or 6, so we dont have much dca's left.

can you implement a dca cooldown, that triggers dca not to buy after a dca buy for x mins.

jess

**Comment by mkarulin** [comment]
+1 this would save lots of money on sudden 10-15% drops

**Comment by GeorgeBIV** [comment]
This, combined with an option to sell back a % of DCA bag (with the option to hold a certain percentage) over x period would be huge.

**Comment by CryptoEU** [comment]
Just a simple timeout is really needed.

If you check some coins and trades, with just a 5 min timeout, you would save tons of money.

**Comment by louisswart** [comment]
I'm using EMASPREAD(0.2) alowing 5 DCA buys and get multiple buys (for example see log and graph below).

The dca cooldown time is a good idea. 

I also think the trigger action should be latched and not allowed to trigger again before it has not first gone false again. Its good to keep trying while "You are not allowed to buy  ..." as funds could become available, but once bought it should not try gain.

```
2017-12-14 10:55:19 INFO ProfitTrailerService - Buy order for BTC-NXS -- {"OrderId":"217e3119-59b7-428a-9364-2ce166d23560","MarketName":"BTC-NXS","MarketCurrency":"NXS","BuyOrSell":"Buy","OrderType":"LIMIT","Quantity":301.4627675610492,"Rate":0.00016585}
2017-12-14 10:55:24 INFO ProfitTrailerService - Get order information -- {"AccountId":null,"OrderUuid":"217e3119-59b7-428a-9364-2ce166d23560","Exchange":"BTC-NXS","Type":"LIMIT_BUY","Quantity":301.46276756,"QuantityRemaining":0.00000000,"Limit":0.00016585,"Reserved":0.04999759,"ReserveRemaining":0.04999759,"CommissionReserved":0.00012499,"CommissionReserveRemaining":0.00000000,"CommissionPaid":0.00012499,"Price":0.04999759,"PricePerUnit":0.00016584000000000000,"Opened":"2017-12-14T08:55:21.247","Closed":"2017-12-14T08:55:21.497","IsOpen":false,"Sentinel":"e243e30d-a81e-4868-a8ec-1d111dc6fea0","CancelInitiated":false,"ImmediateOrCancel":true,"IsConditional":false,"Condition":"NONE","ConditionTarget":null}
2017-12-14 10:55:24 INFO DCAStrategyRunner - DCA - EMASPREAD bought -- currency pair BTC-NXS - current profit -20.341018251681074 - buy_trigger -5.0 - balance 240.14216416
2017-12-14 10:56:03 INFO DCAStrategyRunner - You are not allowed to buy BTC-NXS. Buy will exceed min buy balance
2017-12-14 10:58:25 INFO ProfitTrailerService - DCA Heartbeat
2017-12-14 11:01:09 INFO DCAStrategyRunner - You are not allowed to buy BTC-NXS. Buy will exceed min buy balance
2017-12-14 11:03:36 INFO ProfitTrailerService - Cache Heartbeat
2017-12-14 11:04:58 INFO ProfitTrailerService - Normal Heartbeat
2017-12-14 11:06:13 INFO DCAStrategyRunner - You are not allowed to buy BTC-NXS. Buy will exceed min buy balance
2017-12-14 11:08:27 INFO ProfitTrailerService - DCA Heartbeat
2017-12-14 11:11:17 INFO DCAStrategyRunner - You are not allowed to buy BTC-NXS. Buy will exceed min buy balance
2017-12-14 11:13:39 INFO ProfitTrailerService - Cache Heartbeat
2017-12-14 11:14:58 INFO ProfitTrailerService - Normal Heartbeat
2017-12-14 11:16:20 INFO DCAStrategyRunner - You are not allowed to buy BTC-NXS. Buy will exceed min buy balance
2017-12-14 11:18:35 INFO ProfitTrailerService - DCA Heartbeat
2017-12-14 11:21:20 INFO DCAStrategyRunner - You are not allowed to buy BTC-NXS. Buy will exceed min buy balance
2017-12-14 11:23:40 INFO ProfitTrailerService - Cache Heartbeat
2017-12-14 11:24:59 INFO ProfitTrailerService - Normal Heartbeat
2017-12-14 11:26:26 INFO DCAStrategyRunner - You are not allowed to buy BTC-NXS. Buy will exceed min buy balance
2017-12-14 11:28:38 INFO ProfitTrailerService - DCA Heartbeat
2017-12-14 11:31:26 INFO DCAStrategyRunner - You are not allowed to buy BTC-NXS. Buy will exceed min buy balance
2017-12-14 11:33:41 INFO ProfitTrailerService - Cache Heartbeat
2017-12-14 11:35:01 INFO ProfitTrailerService - Normal Heartbeat
2017-12-14 11:36:30 INFO DCAStrategyRunner - You are not allowed to buy BTC-NXS. Buy will exceed min buy balance
2017-12-14 11:38:41 INFO ProfitTrailerService - DCA Heartbeat
```

![capture2](https://user-images.githubusercontent.com/29467849/34014863-505674ae-e125-11e7-9f19-e178ec2d865b.JPG)

**Comment by louisswart** [comment]
Want to suggest that the DCA Cooldown time starts after the last buy, not only the last DCA buy, but also including the initial buy.

**Reason:**
While a pair is in an uptrend PT buy and then sell when it drops the trailing profit %. This might repeat several times in the uptrend until the last buy near the peak because the buy strategy is still true on the turnaround. The pair then enters a sudden downtrend, DCA is then triggered because the DCA buy strategy is also still true. Thus, 2 buys were done near the peak in a short amount of time before the buy strategy could turn false. If the DCA Cooldown time is relative to the last buy, instead of only the last DCA buy, then the mentioned condition can be prevented.
